src/components/App.tsx

```diff
@@ -262,7 +263,35 @@ export class App extends React.Component<any, AppState> {
                 sceneAppState || getDefaultAppState(),
                 { scrollToContent: true },
               );
-              elements = restoredState.elements;
+              // Perform reconciliation - in collaboration, if we encounter
+              // elements with more staler versions than ours, ignore them
+              // and keep ours.
+              if (elements == null || elements.length === 0) {
+                elements = restoredState.elements;
+              } else {
+                // create a map of ids so we don't have to iterate
+                // over the array more than once.
+                const elementMap = elements.reduce(
+                  (
+                    acc: { [key: string]: ExcalidrawElement },
+                    element: ExcalidrawElement,
+                  ) => {
+                    acc[element.id] = element;
+                    return acc;
+                  },
+                  {},
+                );
+                // Reconcile
+                elements = restoredState.elements.map(element => {
```

Comment:
do you want to consider concating elements that exist locally but not remotely? like in App.tsx line 418 https://github.com/excalidraw/excalidraw/compare/master...petehunt:firebase-2#diff-c8a101915dc6871f1f3d7ad2208d8655R418